lib/new_relic/agent/instrumentation/stripe_subscriber.rb

```diff
@@ -0,0 +1,46 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic
+  module Agent
+    module Instrumentation
+      class StripeSubscriber
+        DEFAULT_DESTINATIONS = AttributeFilter::DST_SPAN_EVENTS
+        EVENT_ATTRIBUTES = %i[http_status method num_retries path request_id].freeze
+
+        def state
+          NewRelic::Agent::Tracer.state
+        end
+
+        def start_segment(event)
+          return unless state.is_execution_traced?
+
+          segment = Tracer.start_segment(name: metric_name(event))
+          event.user_data[:newrelic_segment] = segment
+        rescue => e
+          NewRelic::Agent.logger.debug("Error starting New Relic Stripe segment: #{e}")
```

Comment:
I think this event warrants something more severe than `debug`.

Reply:
I guess if this fails it means something is pretty wrong here. Agree, I've made this an `error` level. https://github.com/newrelic/newrelic-ruby-agent/pull/2180/commits/9c8998032d130be04257693e0cf0a33fce2acd7d